src/lib.rs

```diff
@@ -72,42 +72,89 @@
 //! }
 //! ```
 
-extern crate bytes;
-#[macro_use]
+macro_rules! if_runtime {
+    ($($i:item)*) => ($(
+        #[cfg(all(
+            feature = "reactor",
```

Comment:
This seems like it would be a bit confusing. Perhaps `threadpool` and `current_thread` should *imply* runtime, and thus depend on `reactor` and `timer`?

The thought is that you can depend on those crates directly if you want the actual executors.

Reply:
Actually, another option is that the runtime conditionally adds timer and reactor support **if** those feature flags are enabled...

Reply:
How about now?